code/modules/mob/living/brain/brain_item.dm

```diff
@@ -575,10 +576,9 @@
 	add_trauma_to_traumas(actual_trauma)
 	if(owner)
 		actual_trauma.owner = owner
-		if(SEND_SIGNAL(owner, COMSIG_CARBON_GAIN_TRAUMA, trauma, resilience) & COMSIG_CARBON_BLOCK_TRAUMA)
+		if((SEND_SIGNAL(owner, COMSIG_CARBON_GAIN_TRAUMA, trauma, resilience) & COMSIG_CARBON_BLOCK_TRAUMA) || !actual_trauma.on_gain())
 			qdel(actual_trauma)
 			return FALSE
```

Comment:
if this is the main "action" of the proc it's easier to read if its on its own line (even if it is duplicated)
```suggestion
		if(SEND_SIGNAL(owner, COMSIG_CARBON_GAIN_TRAUMA, trauma, resilience) & COMSIG_CARBON_BLOCK_TRAUMA)
			qdel(actual_trauma)
			return FALSE
		if(!actual_trauma.on_gain())
			qdel(actual_trauma)
			return FALSE
```